gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/IllegalMethodCalledDuringTaskExecution.java

```diff
@@ -55,7 +60,108 @@
         the tasks become independent and can execute in parallel.
         """)
 public final class IllegalMethodCalledDuringTaskExecution extends GradleGuideBugChecker
-        implements BugChecker.MethodTreeMatcher {
+        implements BugChecker.MethodInvocationTreeMatcher {
+    private static final LoadingCache<CompilationUnitTree, MethodCallGraph> callGraphCache =
+            Caffeine.newBuilder().maximumSize(1_000).weakKeys().softValues().build(MethodCallGraph::new);
```

Comment:
We should put this in the method call graph - if multiple bugcheckers want access to the method call graph for the same compilation unit, they should use the same cache.